src/components/EResourceSections/AcquisitionOptions/AcquisitionOptions.js

```diff
@@ -6,55 +6,74 @@ import { Accordion, Badge, MultiColumnList, NoValue, Spinner } from '@folio/stri
 import {
   EResourceType,
   isPackage,
-  TitleOnPlatformLink
+  TitleOnPlatformLink,
+  usePrevNextPagination
 } from '@folio/stripes-erm-components';
 import AddToBasketButton from '../../AddToBasketButton';
 import Coverage from '../../Coverage';
 import EResourceKB from '../../EResourceKB';
 
-import { isExternal } from '../../utilities';
-import { resourceClasses } from '../../../constants';
-
-class AcquisitionOptions extends React.Component {
-  static propTypes = {
-    data: PropTypes.shape({
-      areEntitlementOptionsLoading: PropTypes.bool,
-      entitlementOptions: PropTypes.arrayOf(PropTypes.object),
-      entitlementOptionsCount: PropTypes.number,
-      eresource: PropTypes.shape({
-        name: PropTypes.string,
-      }),
+import { isExternal, parseMclPageSize } from '../../utilities';
+import { resourceClasses, ENTITLEMENT_OPTIONS_PAGINATION_ID } from '../../../constants';
+import { useAgreementsSettings } from '../../../hooks';
+
+const propTypes = {
+  data: PropTypes.shape({
+    areEntitlementOptionsLoading: PropTypes.bool,
+    entitlementOptions: PropTypes.arrayOf(PropTypes.object),
+    entitlementOptionsCount: PropTypes.number,
+    eresource: PropTypes.shape({
+      name: PropTypes.string,
     }),
-    handlers: PropTypes.shape({
-      onEResourceClick: PropTypes.func,
-      onNeedMoreEntitlementOptions: PropTypes.func,
-    }),
-    id: PropTypes.string,
-  };
-
-  getName = (eresource = {}) => {
+  }),
+  handlers: PropTypes.shape({
+    onEResourceClick: PropTypes.func,
+    onNeedMoreEntitlementOptions: PropTypes.func,
+  }),
+  id: PropTypes.string,
+  isLoading: PropTypes.bool.isRequired,
+};
+
+const AcquisitionOptions = ({
+  data: { areEntitlementOptionsLoading, entitlementOptions, entitlementOptionsCount },
+  handlers: { onEResourceClick, onNeedMoreEntitlementOptions },
+  id,
+  isLoading
+}) => {
+  const { settings } = useAgreementsSettings();
```

Comment:
This should be ```const settings = useAgreementsSettings();``` as the parseMCLPageSize expects the full settings object. (Same issue elsewhere)